prow/spyglass/lenses/buildlog/buildlog.ts

```diff
@@ -121,32 +144,67 @@ async function loadLine(artifact: string, line: number): Promise<boolean> {
   return false;
 }
 
-async function handleHash(): Promise<void> {
+function parseHash(): [string, number, number]|null {
   const hash = location.hash.substr(1);
   const colonPos = hash.lastIndexOf(':');
   if (colonPos === -1) {
-    return;
+    return null;
   }
   const artifact = hash.substring(0, colonPos);
-  const lineNum = Number(hash.substring(colonPos + 1));
-  if (isNaN(lineNum)) {
+  const lineRange = hash.substring(colonPos + 1);
+  const hyphenPos = lineRange.lastIndexOf('-');
+
+  let startNum;
+  let endNum;
+
+  if (hyphenPos > 0 ) {
+    startNum = Number(lineRange.substring(0, hyphenPos));
+    endNum = Number(lineRange.substring(hyphenPos + 1));
+  } else {
+    startNum = Number(lineRange);
+    endNum = startNum;
+  }
+  if (isNaN(startNum) || isNaN(endNum)) {
+    return null;
+  }
+  if (endNum < startNum) {
+    [startNum, endNum] = [endNum, startNum];
+  }
```

Comment:
Is this to catch instances where the URL has the start/end numbers swapped? Might be good to comment that in here.

Reply:
Done